File: SimpleHabit/app/src/main/java/com/me/simplehabit/network/SimpleHabitDataAgent.java
package com.me.simplehabit.network;

import com.me.simplehabit.delegates.CategoryProgramDelegate;
import com.me.simplehabit.delegates.CurrentProgramResponseDelegate;
import com.me.simplehabit.delegates.TopicResponseDelegate;

public interface SimpleHabitDataAgent {

    void getCurrentProgram(String token, int page, CurrentProgramResponseDelegate delegate);
    void getTopics(String token, int page, TopicResponseDelegate delegate);
    void getCategoriesAndPrograms(String token, int page, CategoryProgramDelegate delegate);
}

File: SimpleHabit/app/src/main/java/com/me/simplehabit/network/SimpleHabitDataAgent.kt
package com.me.simplehabit.network

import com.me.simplehabit.delegates.CategoryProgramDelegate
import com.me.simplehabit.delegates.CurrentProgramResponseDelegate
import com.me.simplehabit.delegates.TopicResponseDelegate

interface SimpleHabitDataAgent {

    fun getCurrentProgram(token: String, page: Int, delegate: CurrentProgramResponseDelegate)
    fun getTopics(token: String, page: Int, delegate: TopicResponseDelegate)
    fun getCategoriesAndPrograms(token: String, page: Int, delegate: CategoryProgramDelegate)
}
